src/core/core.controller.js

```diff
@@ -285,19 +286,24 @@ class Chart {
 		return this;
 	}
 
-	resize(silent) {
+	resize(silent, width, height) {
 		const me = this;
 		const options = me.options;
 		const canvas = me.canvas;
-		const aspectRatio = (options.maintainAspectRatio && me.aspectRatio) || null;
-		const oldRatio = me.currentDevicePixelRatio;
+		const aspectRatio = options.maintainAspectRatio && me.aspectRatio;
 
+		if (height === undefined) {
```

Comment:
should we check `width === undefined || height === undefined` ?

Reply:
Maybe. I just tested the last parameter to detect missing params.